test/laplace.jl

```diff
@@ -0,0 +1,178 @@
+function generate_data()
+    Random.seed!(1)
+    X = range(0, 23.5; length=48)
+    fs = @. 3 * sin(10 + 0.6X) + sin(0.1X) - 1
+    # invlink = normcdf
+    invlink = logistic
+    ps = invlink.(fs)
+    Y = [rand(Bernoulli(p)) for p in ps]
+    return X, Y
+end
+
+dist_y_given_f(f) = Bernoulli(logistic(f))
+
+function build_latent_gp(theta)
+    variance = softplus(theta[1])
+    lengthscale = softplus(theta[2])
+    kernel = variance * with_lengthscale(SqExponentialKernel(), lengthscale)
+    return LatentGP(GP(kernel), dist_y_given_f, 1e-8)
+end
+
+function optimize_elbo(
+    build_latent_gp,
+    theta0,
+    xs,
+    ys,
+    optimizer,
+    optim_options;
+    newton_warmstart=true,
+    newton_callback=nothing,
+)
+    objective = build_laplace_objective(
+        build_latent_gp, xs, ys; newton_warmstart, newton_callback
+    )
+    objective_grad(θ) = only(Zygote.gradient(objective, θ))
+
+    training_results = Optim.optimize(
+        objective, objective_grad, theta0, optimizer, optim_options; inplace=false
+    )
+
+    lf = build_latent_gp(training_results.minimizer)
+    f_post = posterior(LaplaceApproximation(; f_init=objective.f), lf(xs), ys)
+    return f_post, training_results
+end
+
+@testset "Gaussian" begin
+    # should check for convergence in one step, and agreement with exact GPR
+end
+
+@testset "gradients" begin
+    X, Y = generate_data()
+    @testset "laplace_lml" begin
+        theta0 = rand(2)
+        function objective(theta)
+            lf = build_latent_gp(theta)
+            lml = laplace_lml(lf(X), Y)
+            return -lml
+        end
+        fd_grad = only(FiniteDifferences.grad(central_fdm(5, 1), objective, theta0))
+        ad_grad = only(Zygote.gradient(objective, theta0))
+        @test ad_grad ≈ fd_grad
+    end
+end
+
+@testset "chainrule" begin
+    xs = [0.2, 0.3, 0.7]
+    ys = [1, 1, 0]
+    L = randn(3, 3)
```

Comment:
should probably fix a seed (the alternative would be to increase the isapprox rtol)